packages/wasm/crate/src/storage.rs

```diff
@@ -386,6 +389,63 @@ impl IndexedDBStorage {
             Err(WasmError::Anyhow(Error::msg("could not find reserves")))
         }
     }
+
+    pub async fn get_delegation_assets(&self) -> WasmResult<BTreeMap<Id, DelegationToken>> {
+        let idb_tx = self.db.transaction_on_one(&self.constants.tables.assets)?;
+        let store = idb_tx.object_store(&self.constants.tables.assets)?;
+        let all_metadata = store
+            .get_all()?
+            .await?
+            .into_iter()
+            .map(serde_wasm_bindgen::from_value)
+            .collect::<Result<Vec<Metadata>, _>>()?;
+
+        let mut assets: BTreeMap<Id, DelegationToken> = BTreeMap::new();
+
+        for metadata in all_metadata {
+            if let Ok(token) = DelegationToken::try_from(metadata.clone()) {
+                assets.insert(metadata.id(), token);
+            }
+        }
+
+        Ok(assets)
+    }
+
+    pub async fn get_notes_for_voting(
+        &self,
+        address_index: Option<keys::v1::AddressIndex>,
+        votable_at_height: u64,
+    ) -> WasmResult<Vec<(SpendableNoteRecord, IdentityKey)>> {
+        let delegation_assets = self.get_delegation_assets().await?;
+
+        let all_notes = self
+            .get_notes(NotesRequest {
+                include_spent: true,
+                address_index,
+                asset_id: None,
+                amount_to_spend: None,
+            })
+            .await?;
```

Comment:
You are loading all notes into memory, this could be an issue in the future given that WASM has memory limitations

Reply:
Perhaps it would be better if we could pass `votable_notes` to `TransactionPlannerRequest.DelegatorVote`

Reply:
The memory note is definitely valid. It's something we'll need some time to think about. Multiple parts of the planner rely upon having all the notes in memory.